### PR TITLE
Proto parser demo generator

### PR DESCRIPTION
**Pull request by alexander871015**
Written the implementation of proto parser generator which takes the base text proto and generates any number of similarly formed text protos with different values to show proto parser's ability for large scale changes.

### CODE REVIEW

File: proto_parser/demo/demo_generator.cc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  if (rand() % 2) {
+    ui_element_path_predicate->DeleteField(1);
+  }
+  
+  return message->PrintToTextProto();
+}
+
+} // namespace
+
+int main() {
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_generator.cc:80]: Add comment about what this main is doing.

Rename the file to demo_generator_main.cc

File: proto_parser/demo/demo_generator.cc [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/*
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_generator.cc:1]: demo_sample_generator_main.cc

File: proto_parser/demo/BUILD [hunk 3]
```diff
@@ -0,0 +1,10 @@
+cc_binary(
+  name = "demo_generator",
```
Comment (sarahchen-google) [inline proto_parser/demo/BUILD:2]: keep the name the same as src name.

demo_sample_generator_main

File: proto_parser/demo/demo_sample_generator_main.cc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+  ifs.open("demo/proto_texts/original_proto_text.txt");
+
+  assert(ifs.good());
+
+  while (ifs.good()) {
+    std::string tmp;
+    std::getline(ifs, tmp);
+    text_proto += tmp + "\n";
+  }
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_sample_generator_main.cc:95]: Move this into a function 

std::string ReadTextProto(absl::string_view file)

File: proto_parser/demo/demo_sample_generator_main.cc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+using ::wireless_android_play_analytics::ProtoValue;
+using ::wireless_android_play_analytics::MessageValue;
+using ::wireless_android_play_analytics::PrimitiveValue;
+using ::wireless_android_play_analytics::EventPredicate;
+using ::wireless_android_play_analytics::ProtoParser;
+
+namespace {
+
+std::string Generate(MessageValue* message) {
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_sample_generator_main.cc:32]: MessageValue* ->const MessageValue&

Comment (sarahchen-google) [inline proto_parser/demo/demo_sample_generator_main.cc:32]: Generate() -> GenerateRandomSamples()

File: proto_parser/src/BUILD [hunk 6]
```diff
@@ -2,26 +2,27 @@ package(default_visibility = ["//visibility:public"])
...
-    hdrs = ["proto_parser.h"],
-    deps = ["proto_value",
-            "primitive_value",
-            "message_value"]
-)
-
 cc_library(
     name = "proto_value",
     srcs = ["proto_value.cc"],
-    hdrs = ["proto_value.h", "proto_parser.h", "message_value.h", "primitive_value.h"],
```
Comment (sarahchen-google) [inline proto_parser/src/BUILD:17]: Merge the branch with master to get rid of these changes.

File: proto_parser/demo/demo_sample_generator_main.cc (hunk 5 above)
Comment (alexander871015) [reply proto_parser/demo/demo_sample_generator_main.cc:32]: Here we are in fact modifying the MessageValue.

File: proto_parser/demo/proto_texts/original_proto_text.txt [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# comment
```
Comment (sarahchen-google) [inline proto_parser/demo/proto_texts/original_proto_text.txt:1]: change the file to 

original_proto_text.textproto

File: proto_parser/demo/demo_sample_generator_main.cc [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+int main() {
+  EventPredicate event_predicate;
+  ProtoParser parser(
+      ReadTextProtoFromStream("demo/proto_texts/original_proto_text.txt"));
+
+  for(int i = 0; i < 10; ++i) {
+    std::unique_ptr<ProtoValue> message = parser.Create(event_predicate);
+    std::ofstream outs;
+    outs.open("demo/proto_texts/proto_text(" + std::to_string(i) +").txt");
+    outs << GenerateRandomSamples(static_cast<MessageValue*>(message.get()));
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_sample_generator_main.cc:109]: Move this to function

void WriteTextproto(absl::string_view file, const MessageValue&)

File: proto_parser/demo/demo_sample_generator_main.cc [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+using ::wireless_android_play_analytics::ProtoValue;
+using ::wireless_android_play_analytics::MessageValue;
+using ::wireless_android_play_analytics::PrimitiveValue;
+using ::wireless_android_play_analytics::EventPredicate;
+using ::wireless_android_play_analytics::ProtoParser;
+
+namespace {
+
+std::string GenerateRandomSamples(MessageValue* message) {
```
Comment (sarahchen-google) [inline proto_parser/demo/demo_sample_generator_main.cc:32]: MessageValue*  -> const MessageValue&

Comment (alexander871015) [reply proto_parser/demo/demo_sample_generator_main.cc:32]: As stated in https://github.com/googleinterns/sarahchen-intern-2020/pull/36#discussion_r454620936, we are actually modifying the message in this function.

**Pull request merged**